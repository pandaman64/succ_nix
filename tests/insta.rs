use std::fs;

#[test]
fn test_insta() {
    tracing_subscriber::fmt::init();

    insta::glob!("corpus/*.nix", |path| {
        let input = fs::read_to_string(path).unwrap();
        let (ty, sol, sink) = succ_nix::run(&input, 10000);

        if !sink.is_error() {
            insta::assert_snapshot!(sol.map_ty(&ty).to_string());
        }
    });
}
