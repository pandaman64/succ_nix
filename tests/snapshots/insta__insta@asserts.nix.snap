---
source: tests/insta.rs
expression: sol.map_ty(&ty).to_string()
input_file: tests/corpus/asserts.nix

---
({ lib = { elem = (any -> (any -> bool)); ... = any; }; ... = any; } -> { assertMsg = (bool -> (any -> any)); assertOneOf = (any -> (any -> (any -> any))); ... = âˆ…; })
